lightsaber/processor/src/test/java/com/joom/lightsaber/processor/validation/FactoryParserTest.kt

```diff
@@ -69,4 +69,26 @@ class FactoryParserTest {
           "Dependency(type=java.lang.String, qualifier=null)"
     )
   }
+
+  @Test
+  fun test_check_fails_when_factory_has_misconfigured_methods() {
+    integrationTestRule.assertInvalidProject(
+      sourceCodeDir = "factory_with_misconfigured_inheritance",
+      message = "Class test_case_projects.factory_parser.factory_with_misconfigured_inheritance.Bar must have a constructor annotated with @Factory.Inject"
```

Comment:
Is it possible to defer a more meaningful error for this case?

Reply:
Hm, actually it would be the same as if the factory itself had method `fun createBar(): Bar`. nwm then